wordpress-mobile/WordPress-iOS
# Implement UISwipeActions / Remove MGSwipeTableCell

Now that we've bumped our base SDK up to iOS 11, we can drop MGSwipeTableCell and use native UISwipeActions instead. This PR does just that. Apologies that it's a little long, but the majority of the changes are just cookie cutter across the different types of notifications and their tests.

![before-after-swipes](https://user-images.githubusercontent.com/4780/61718299-9cbffb00-ad5a-11e9-8f8e-189de768d0eb.png)

A few notes:

* I was kind of confused by our existing setup for these actions. From what I can tell, every notification action type (such as Like Comment, Like Post, Reply, Spam) implemented a method returning a button, but it seems as though we were only using Approve and Trash (and mark as read, but that's separate). I've implemented contextual actions for all of them for consistency, but I'm not sure they're all needed right now. When testing, please ensure the actions still show up in places like the comment detail screen.
* You may notice the text size by default is a little smaller than previously. It's not configurable by us, but it does support dynamic type, so that should be fine.
* It also doesn't appear to support accessibility labels or hints unfortunately, but system swipe actions are accessible whereas I don't think MGSwipeTableCell was. If you're interested in testing this, with Voice Over active: tap a row to select it, make a two finger rotate gesture to bring up the 'rotor', and change it to Actions, then swipe up and down with one finger. It should read the available actions such as Approve, Trash, etc.

**To test:**

* Build and run
* Check unit tests pass and look ok (I had to update them)
* Ensure that you can do the same things you could previously do with notifications:
- [ ] Swipe to mark as read
- [ ] Swipe to mark as unread
- [ ] Swipe to approve
- [ ] Swipe to unapprove
- [ ] Swipe to trash
* Check you can undo a trash
* Check you can still trigger notification actions from other places: comment notification detail views, and push notification actions

Update release notes:

(I don't think this is directly user facing as it should be largely invisible)
